forge/routes/api/project.js

```diff
@@ -394,20 +394,25 @@ module.exports = async function (app) {
                     env: request.body.settings.env
                 }
             }
-            const newSettings = app.db.controllers.ProjectTemplate.validateSettings(bodySettings, request.project.ProjectTemplate)
-            if (newSettings.httpNodeAuth?.type === 'flowforge-user') {
-                const teamType = await request.project.Team.getTeamType()
-                if (teamType.properties.features?.teamHttpSecurity === false) {
-                    reply.code(400).send({ code: 'invalid_request', error: 'FlowForge User Authentication not available for this team type' })
-                    return
+            try {
+                const newSettings = app.db.controllers.ProjectTemplate.validateSettings(bodySettings, request.project.ProjectTemplate)
+                if (newSettings.httpNodeAuth?.type === 'flowforge-user') {
+                    const teamType = await request.project.Team.getTeamType()
+                    if (teamType.properties.features?.teamHttpSecurity === false) {
+                        reply.code(400).send({ code: 'invalid_request', error: 'FlowForge User Authentication not available for this team type' })
+                        return
+                    }
                 }
-            }
 
-            // Merge the settings into the existing values
-            const currentProjectSettings = await request.project.getSetting(KEY_SETTINGS) || {}
-            const updatedSettings = app.db.controllers.ProjectTemplate.mergeSettings(currentProjectSettings, newSettings)
+                // Merge the settings into the existing values
+                const currentProjectSettings = await request.project.getSetting(KEY_SETTINGS) || {}
+                const updatedSettings = app.db.controllers.ProjectTemplate.mergeSettings(currentProjectSettings, newSettings)
 
-            changesToPersist.settings = { from: currentProjectSettings, to: updatedSettings }
+                changesToPersist.settings = { from: currentProjectSettings, to: updatedSettings }
+            } catch (err) {
+                reply.code(400).send({ code: 'env_var_validation', error: `${err.message}` })
+                return
+            }
```

Comment:
The try catch is too course. It will catch other errors raised in `app.db.controllers.ProjectTemplate.validateSettings` and all of the other external code calls and report them with code `env_var_validation`

Example: Sending an update with a bad module entry results a mis-matched error:

![image](https://github.com/flowforge/flowforge/assets/44235289/57607664-15bb-41ba-96a2-e69d3975f9cd)


Ideally we would have a lib for generating properly formed errors (and for checking/appending the elements like `code` and `error` props) but in the absence of that, the controller should either throw well formatted errors (that we add to the response) or we should inspect the error before declaring it an Env Var Validation error

Reply:
I had to wrap all the uses of `newSettings`